components/cli/ballerina/docker_bal_executor.go

```
@@ -137,12 +137,117 @@ func (baleExecutor *DockerBalExecutor) Build(fileName string, iName []byte) erro
 	return nil
 }
 
+// Run executes ballerina run when ballerina is not installed.
+func (balExecutor *DockerBalExecutor) Run(imageDir string, instanceName string, runningNode *DependencyTreeNode,
+	envVars []*EnvironmentVariable, tempRunFileName string, dependencyLinks map[string]*DependencyInfo,
+	startDependencies, shareDependencies bool) error {
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		util.ExitWithErrorMessage("Error in determining working directory", err)
```

Please return the error

---

Updated in all places.